### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by cschi0815**
Fixes #SPG-1867

What did you change (functionally and technically)?

Added CodeQl

**Checklist** (remove this list before you submit the PR)
* Are there tests for the new code? 

No new code

* Does the code comply to our code conventions?

No new code

* Does the PR resolve the whole issue?

No just configuration

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "codeQlAct"]
```
Comment (steffektif) [inline .github/workflows/codeql.yml:16]: Why does this trigger on "codeQlAct"?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ,"codeQlAct"]
+  schedule:
+    - cron: '45 7 * * 0'
```
Comment (steffektif) [inline .github/workflows/codeql.yml:21]: Can you translate that cron to something human readable as comment for convenience ?

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+           queries: security-extended
+           config-file: ./.github/codeql/codeql-config.yml
+           languages:  ${{ matrix.language }}
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, Go, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
```
Comment (steffektif) [inline .github/workflows/codeql.yml:55]: I guess we first gonna see if that works before we remove it?

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (cschi0815) [reply .github/workflows/codeql.yml:16]: i can also remove it, if you think its better

Comment (steffektif) [reply .github/workflows/codeql.yml:16]: It's fine, was just curious why you would have it.

**Review by steffektif — APPROVED** [review]

**Pull request merged**